physics/MP/GFDL/v3_2022/gfdl_cloud_microphys_v3_mod.F90

```diff
@@ -0,0 +1,7377 @@
+! terminal velocity for rain, snow, and graupel, Lin et al. (1983)
+! =======================================================================
+
+subroutine term_rsg (ks, ke, q, den, denfac, v_fac, blin, mu, tva, tvb, v_max, const_v, vt)
+
+    implicit none
+
+    ! -----------------------------------------------------------------------
+    ! input / output arguments
+    ! -----------------------------------------------------------------------
+
+    integer, intent (in) :: ks, ke
+
+    logical, intent (in) :: const_v
+
+    real, intent (in) :: v_fac, blin, v_max, mu
+
+    real (kind = r8), intent (in) :: tva, tvb
+
+    real, intent (in), dimension (ks:ke) :: q, den, denfac
+
+    real, intent (out), dimension (ks:ke) :: vt
+
+    ! -----------------------------------------------------------------------
+    ! local variables
+    ! -----------------------------------------------------------------------
+
+    integer :: k
+
+    if (const_v) then
+        vt (:) = v_fac
+    else
+        do k = ks, ke
+            if (q (k) .lt. qfmin) then
+                vt (k) = 0.0
+            else
+                call cal_pc_ed_oe_rr_tv (q (k), den (k), blin, mu, &
+                    tva = tva, tvb = tvb, tv = vt (k))
+                vt (k) = v_fac * vt (k) * denfac (k)
+                vt (k) = min (v_max, max (0.0, vt (k)))
+            endif
+        enddo
+    endif
+
+end subroutine term_rsg
+
+! =======================================================================
+! melting during sedimentation
+! =======================================================================
+
+subroutine sedi_melt (dts, ks, ke, tz, qv, ql, qr, qi, qs, qg, dz, dp, &
+        vt, r1, tau_mlt, icpk, qflag)
+
+    implicit none
+
+    ! -----------------------------------------------------------------------
+    ! input / output arguments
+    ! -----------------------------------------------------------------------
+
+    integer, intent (in) :: ks, ke
+
+    real, intent (in) :: dts, tau_mlt
+
+    real, intent (in), dimension (ks:ke) :: vt, dp, dz, icpk
+
+    real, intent (inout), dimension (ks:ke) :: qv, ql, qr, qi, qs, qg
+
+    real, intent (inout) :: r1
+
+    real (kind = r8), intent (inout), dimension (ks:ke) :: tz
+
+    character (len = 2), intent (in) :: qflag
+
+    ! -----------------------------------------------------------------------
+    ! local variables
+    ! -----------------------------------------------------------------------
+
+    integer :: k, m
+
+    real :: dtime, sink, zs
+
+    real, dimension (ks:ke) :: q
+
+    real, dimension (ks:ke + 1) :: ze, zt
+
+    real (kind = r8), dimension (ks:ke) :: cvm
+
+    call zezt (ks, ke, dts, zs, dz, vt, ze, zt)
+
+    select case (qflag)
+        case ("qi")
+            q = qi
+        case ("qs")
+            q = qs
+        case ("qg")
+            q = qg
+        case default
+            print *, "gfdl_mp: qflag error!"
```

Comment:
Should one pass down and set ccpp_err_msg and ccpp_err_flag and return? Or is this error not that egregious?